tests/extra_python_package/test_files.py

```diff
@@ -0,0 +1,245 @@
+# -*- coding: utf-8 -*-
+import subprocess
+import sys
+import os
+import tarfile
+import contextlib
+import zipfile
+
+# These tests must be run explicitly
+# They require CMake 3.15+ (--install)
+
+DIR = os.path.abspath(os.path.dirname(__file__))
+MAIN_DIR = os.path.dirname(os.path.dirname(DIR))
+
+
+main_headers = {
+    "include/pybind11/attr.h",
+    "include/pybind11/buffer_info.h",
+    "include/pybind11/cast.h",
+    "include/pybind11/chrono.h",
+    "include/pybind11/common.h",
+    "include/pybind11/complex.h",
+    "include/pybind11/eigen.h",
+    "include/pybind11/embed.h",
+    "include/pybind11/eval.h",
+    "include/pybind11/functional.h",
+    "include/pybind11/iostream.h",
+    "include/pybind11/numpy.h",
+    "include/pybind11/operators.h",
```

Comment:
Isn't this what `check-manifest` does, already, checking which files are included in the source distribution?

Reply:
This is a lot more powerful than check-manifest; this ensures every file is there. We don't really need check-manifest now, since we check file-by-file; it mostly overlaps. Check-manifest only compares git files, so it can't tell you if `pybind11/include/*` or `pybind11/share/*` are there. It also can't check the wheel, which I do later. :)